cranelift/codegen/src/isa/riscv64/inst/encode.rs

```diff
@@ -468,7 +468,7 @@ pub fn encode_ciw_type(op: CiwOp, rd: WritableReg, imm: u8) -> u16 {
     let mut imm_field = 0;
     imm_field |= ((imm >> 1) & 1) << 0;
     imm_field |= ((imm >> 0) & 1) << 1;
-    imm_field |= ((imm >> 4) & 7) << 2;
+    imm_field |= ((imm >> 4) & 15) << 2;
```

Comment:
Do we have `emit_tests.rs` for riscv64? I would kind of expect some updated tests in there that exercise this, rather than just the checked in fuzz test.

Reply:
I've been using the disassembled section in the compile tests as a check that the encodings are correct. However, we don't have many compile tests specifically for C instructions. Most of them test only with the Base ISA and don't enable the other extensions so they won't be emitting this instruction.

Reply:
Right, I guess before merging this I would like to have either a test in `emit_tests.rs` that does some loop over min/max/etc values for this type of immediate and encodes each of them or something else like that in spirit (e.g. a series of `c.addi4spn` instructions with various min/max/etc values in a filetest).

Reply:
I've added a test for both min and max values as well as one that sets a bit in each subfield in the immediate of the instruction. I've also added tests for out of bound values.

I added them into `zca.clif` instead of `emit_tests.rs`, since its slightly awkward to add these there because they are encoding only changes we don't actually know the name of the instruction in cranelift. So the test would look something like `addi a1, sp, 4` but the result is a 2 byte `c.addisp4n`.